plugins/gstperf.c

```diff
@@ -217,6 +314,12 @@ gst_perf_stop (GstBaseTransform * trans)
 
   gst_perf_clear (perf);
 
+  g_free (perf->bps_window_buffer);
+
+  g_mutex_clear (&perf->bps_mutex);
```

Comment:
You are clearing the mutex, but never initializing it again, this will fail with several transitions.